pytorch_lightning/callbacks/timer.py

```diff
@@ -164,8 +164,9 @@ def on_load_checkpoint(
         self._offset = time_elapsed.get(RunningStage.TRAINING.value, 0)
 
     def _check_time_remaining(self, trainer: "pl.Trainer") -> None:
+        assert self._duration is not None
         should_stop = self.time_elapsed() >= self._duration
-        should_stop = trainer.training_type_plugin.broadcast(should_stop)
+        should_stop = cast(bool, trainer.training_type_plugin.broadcast(should_stop))
```

Comment:
This should not be necessary, the fix should be in broadcast

where the correct signature update is

```python
TBroadcast = TypeVar("TBroadcast")

def broadcast(self, obj: TBroadcast, src: int = 0) -> TBroadcast:
```

Reply:
@four4fish do you recall why the above was changed^?

Reply:
@carmocca updated

Reply:
LGTM, resolving